### DISCUSSION TITLE
How to tune hyper-paramter while using one-shot algorithms.

### DISCUSSION THREAD

**Discussion by songlei00** [microsoft/nni#4220] [Q&A] [answered]
- NNI version: 2.3
- Training service: local
- OS: ubuntu18.04
- I use PyTorch. 

I found a possible solution in [Can we do hyperparameter optimization while doing NAS?](https://github.com/microsoft/nni/issues/2591). However, when we use HPO in NNI, we should use function ```nni.report_final_result``` to record the metric during training, as follows:

``` python
for epoch in range(args.epochs):
  metric = run_model()
  nni.report_final_result(metric)
```

One-shot algorithms like Darts package almost all details. I don't know how to add ```nni.report_final_result```. If I don't add it, the web UI is empty, even if the trial is succesed, like the following picture. 

![QQ截图20210806182508](https://user-images.githubusercontent.com/44307424/128496811-02a909a5-6a1b-4790-be5f-67df063b2e16.png)

**Comment by ultmaster** [comment, marked as answer]
You should report the result for the final accuracy of retrained searched model. This means you should combine the search script with retrain script, report the final test accuracy from retrain script.
